packages/protocol/contracts/layer1/surge/SurgeTimelockController.sol

```diff
@@ -52,9 +58,14 @@ contract SurgeTimelockedController is TimelockController {
         super.executeBatch(targets, values, payloads, predecessor, salt);
     }
 
-    /// @dev Returns `true` if an L2 block has not been verified in the last 1 day
+    function updateMinLivenessStreak(uint64 _minLivenessStreak) external onlyRole(TIMELOCK_ADMIN_ROLE) {
+        minLivenessStreak = _minLivenessStreak;
```

Comment:
Should we protect this with `_isLivenessDisrupted()`? Or else, can't a malicious upgrader modify `minLivenessStreak` to zero to bypass the liveness streak check?

Also, shouldn't this function be protected via `minDelay`? Or maybe `onlyRole(TIMELOCK_ADMIN_ROLE)` is doing that in some way, not sure.

Reply:
> Also, shouldn't this function be protected via minDelay? 

I guess what we want to protect `minLivenessStreak` in same way as we protect upgrading of `minDelay` https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/governance/TimelockController.sol#L449

Reply:
Hmm, actually, maybe we are already doing this via `onlyRole(TIMELOCK_ADMIN_ROLE)`